lib/src/main/java/com/springboot/webpush/common/api/Notification.java

```diff
@@ -13,14 +13,14 @@
  * services.
  */
 public class Notification {
-    private final String              title;
-    private final String              body;
-    private final Map<String, Object> data;
-    private final List<Integer>       vibrate;
-    private final List<Action>        actions;
-    private final String              icon;
-    private final String              image;
-    private final boolean             requireInteraction;
+    private final String                   title;
+    private final String                   body;
+    private final Map<String, Object>      data;
```

Comment:
We could more strongly type this data field. As far as I can tell, this is kind of free-form JSON config, which is up to the PWA to deal with. We're using Angular's service worker/PWA implementation, which does optionally expect onActionClick (and it's this property on data that actually gets the PWA to open on click). If there were other custom data we wanted to add later to make everything more structured we could also add that to the potential new Bean (instead of using a generic Map)

Reply:
I'm not exactly sure "what" the bean would be... so if you want to propose something. great... Otherwise, it will stay until I'm smarter!

Reply:
I pushed something up. Let me know what you think